python/taichi/lang/matrix.py

```diff
@@ -415,13 +415,19 @@ def E(x, y):
 
     @staticmethod
     def cross(a, b):
-        assert a.n == 3 and a.m == 1
-        assert b.n == 3 and b.m == 1
-        return Matrix([
-            a(1) * b(2) - a(2) * b(1),
-            a(2) * b(0) - a(0) * b(2),
-            a(0) * b(1) - a(1) * b(0),
-        ])
+        if a.n == 3 and a.m == 1 and b.n == 3 and b.m == 1:
+            return Matrix([
+                a(1) * b(2) - a(2) * b(1),
+                a(2) * b(0) - a(0) * b(2),
+                a(0) * b(1) - a(1) * b(0),
+            ])
+
+        elif a.n == 2 and a.m == 1 and b.n == 2 and b.m == 1:
+            return Matrix(a(0) * b(1) - a(1) * b(0))
+
+        else:
+            raise Exception(
+                "CrossProduct only supports 3D vector and 2D vector")
```

Comment:
```suggestion
                "cross product is only supported on 3D and 2D vector")
```

Reply:
Hi Yubing,
Thanks for the quick suggestion. But here I assume "cross product is only supported between pairs of 3D vectors or those of 2D vectors" is more accurate. Just in case people are confused about if they ever tried vec2D.cross(vec3D).
What do you think~

Reply:
Thank for clarify, that sounds like a better expression.